backends/ebpf/runtime/ebpf_runtime_kernel.c

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <unistd.h>     // getopt()
+#include <ctype.h>      // isprint()
+#include <string.h>     // memcpy()
+#include <stdlib.h>     // malloc()
+#include <arpa/inet.h>
+#include <linux/if_packet.h>
+#include <netinet/ether.h>
+#include <sys/time.h>
+#include <net/if.h>
+#include "ebpf_kernel.h"
+#include "ebpf_runtime_kernel.h"
+
+
+#define FILE_NAME_MAX 256
+#define MAX_10_UINT16 5;
+#define PCAPOUT "_out.pcap"
+
+/**
+ * @brief Create a pcap file name from a given base name, interface index,
+ * and suffix. Return value must be deallocated after usage.
+ * @param pcap_base The file base name.
+ * @param index The index of the file, represent an interface.
+ * @param suffix  Filename suffix (e.g., _in.pcap)
+ * @return An allocated string containing the file name.
+ */
+char *generate_pcap_name(const char *pcap_base, int index, const char *suffix) {
+    /* Dynamic string length plus max decimal representation of uint16_t */
+    int file_length = strlen(pcap_base) + strlen(suffix) + MAX_10_UINT16;
+    char *pcap_name = malloc(file_length);
+    int offset = snprintf(pcap_name, file_length,"%s%d%s",
+                    pcap_base, index, suffix);
+    if (offset >= FILE_NAME_MAX) {
+        fprintf(stderr, "Name %s%d%s too long.\n", pcap_base, index, suffix);
+        exit(EXIT_FAILURE);
+    }
+    return pcap_name;
+}
+
+void *run_tcpdump(char *filename, char *interface) {
```

Comment:
I am running tcpdump from the runtime. I could also run it from the python script, although I prefer this version.